burr/core/application.py

```diff
@@ -1194,7 +1217,7 @@ def with_state(self, **kwargs) -> "ApplicationBuilder":
         """
         if self.initializer is not None:
             raise ValueError(
```

Comment:
For the future, we may want to decorate these with something that catches the error and rethrows with help info, but this will do for now.